File: csharp/Structural/10-Facade/Subsystem/BillingService.cs
namespace Facade.Subsystem;

public class BillingService
{
    public decimal GenerateQuote(string category, bool isUrgent)
    {
        var baseRate = category switch
        {
            "Plumbing" => 850m,
            "Electrical" => 1200m,
            "Carpentry" => 600m,
            _ => 750m
        };

        var total = isUrgent ? baseRate * 1.25m : baseRate;
        Console.WriteLine($"  💰 BillingService: Quote generated — N${total:N2}");
        return total;
    }

    public void SendQuote(string jobId, string clientEmail, decimal amount)
    {
        Console.WriteLine($"  💰 BillingService: Quote of N${amount:N2} sent to {clientEmail} for Job #{jobId}.");
    }
}

File: typescript/Structural/10-Facade/Subsystem/BillingService.ts
export class BillingService {
  generateQuote(category: string, isUrgent: boolean): number {
    const rates: Record<string, number> = {
      Plumbing: 850,
      Electrical: 1200,
      Carpentry: 600,
    };

    const base = rates[category] ?? 750;
    const total = isUrgent ? base * 1.25 : base;

    console.log(`  💰 BillingService: Quote generated — N$${total.toFixed(2)}`);
    return total;
  }

  sendQuote(jobId: string, clientEmail: string, amount: number): void {
    console.log(
      `  💰 BillingService: Quote of N$${amount.toFixed(2)} sent to ${clientEmail} for Job #${jobId}.`,
    );
  }
}
